traits/ctraits.pyx

```diff
@@ -404,9 +437,14 @@ cdef object validate_trait_complex(cTrait trait, CHasTraits obj, object name, ob
     raise_trait_error(trait, obj, name, value)
 
 cdef object validate_trait_tuple(cTrait trait, CHasTraits obj, object name, object value):
-    return validate_trait_tuple_check(trait.py_validate[1], obj, name, value)
+    cdef object result =  validate_trait_tuple_check(
+        trait.py_validate[1], obj, name, value
+    )
+    if result is not None:
+        return result
+    raise_trait_error(trait, obj, name, value)
 
-cdef object validate_trait_tuple_check(cTrait trait, CHasTraits obj, object name, object value):
+cdef object validate_trait_tuple_check(object trait, CHasTraits obj, object name, object value):
```

Comment:
`py_validate[1]` turns out to be a tuple sometimes. I don't know enough about traits to know how this works and what it means, but since the C implementation also has `object` I thought it would be appropriate to change to `object` (see https://github.com/enthought/traits/blob/master/traits/ctraits.c#L3378)